### PR TITLE
Avoid `count()`ing `NULL`

### LINKED ISSUE #178: PHP 8.1 compatibility - do not attempt to `count()` `NULL`
Although donrec in general is working fine with PHP 8.0/8.1, in certain conditions we get a wsod.

This is caused by the php function count, that behaves differently with php 8:
https://www.php.net/manual/en/function.count.php
> count() will now throw [TypeError](https://www.php.net/manual/en/class.typeerror.php) on invalid countable types passed to the value parameter.

I have prepared a pull request.

### PR DESCRIPTION
**Pull request by jensschuppe**
Fixes #178 by not trying to `count()` potentially not initialized `$this->_contactIds` in task forms and use `$this->getContactIDs()` instead which has a null-coalesce guard with a default value of `[]`.

### CODE REVIEW

**Comment by Detsieber** [comment]
@jensschuppe #180 is absolutely fine for me, and it's pretty clear that you know your codebase much better than I do!
Therefore, I have closed #179 

When will release 2.3 be published?

**Comment by jensschuppe** [comment]
@bjendres a quick review? And any concerns releasing this as a `2.2.1` point release instead of waiting for a `2.3`?

**Comment by Detsieber** [comment]
I have tested #180 and it works well. And, releasing as 2.2.1 would be perfect!

**Comment by jensschuppe** [comment]
Released with version `2.2.1`.

File: CRM/Donrec/Form/Task/DonrecTask.php [hunk 1]
```diff
@@ -104,14 +104,24 @@ function postProcess() {
...
       CRM_Core_Session::setStatus(E::ts('There are no selectable contributions for these contacts in the selected time period.'), E::ts('Warning'), 'warning');
       $qfKey = $values['qfKey'];
       CRM_Utils_System::redirect(CRM_Utils_System::url('civicrm/contact/search', "_qf_DonrecTask_display=true&qfKey=$qfKey"));
     }else{
       CRM_Core_Session::singleton()->pushUserContext(
-        CRM_Utils_System::url('civicrm/donrec/task', 'sid=' . $result['snapshot']->getId() . '&ccount=' . count($this->_contactIds))
+        CRM_Utils_System::url(
+          'civicrm/donrec/task',
+          'sid=' . $result['snapshot']->getId()
+          . '&ccount=' . count($this - $this->getContactIDs())
```
Comment (pminf) [inline CRM/Donrec/Form/Task/DonrecTask.php:123]: @jensschuppe I think this calculation is not intended and will cause a type error in PHP 8.

Comment (jensschuppe) [reply CRM/Donrec/Form/Task/DonrecTask.php:123]: Thanks for spotting that, I released 2.2.3 with that being fixed.

**Pull request merged**